tests/LazyPropertyTest/Util/AccessScopeCheckerTest.php

```diff
@@ -30,67 +32,102 @@
  *
  * @covers \LazyProperty\Util\AccessScopeChecker
  */
-class AccessScopeCheckerTest extends PHPUnit_Framework_TestCase
+class AccessScopeCheckerTest extends TestCase
 {
-    public function testAllowsAccessFromSameInstance()
+    public function testAllowsAccessFromSameInstance(): void
     {
-        $this->assertNull(AccessScopeChecker::checkCallerScope(['object' => $this], $this, 'backupGlobals'));
+        try {
+            AccessScopeChecker::checkCallerScope(['object' => $this], $this, 'backupGlobals');
+        } catch (\Exception $exception) {
+            $this->fail('Unexpected exception.');
+        }
+        
+        $this->assertTrue(true);
```

Comment:
There should be a way to put an assertion in the `AccessScopeChecker` instead

Reply:
Same applies to all changes in this file: artificially incrementing the assertion counter should only be done as a very last resort

Reply:
I didn't want to introduce too much change in the actual code, but I'd like to go for refactoring the `checkCallerScope()` method to return a boolean instead of throwing exceptions and instead throw the exception from the `__get()` method. That would make testing better as well since we can simply `assertTrue()` and `assertFalse()`. How do you feel about that?

Reply:
Hmm, I see what you mean there. Maybe we'll have to design an integration test instead, but changing the `AccessScopeChecker` will be trouble

Reply:
Yeah, I think that incrementing the assertion count (manually) here is better than adding fake assertions.